cpp/include/cudf/strings/detail/gather.cuh

```diff
@@ -65,67 +73,84 @@ std::unique_ptr<cudf::column> gather(
   rmm::cuda_stream_view stream,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource())
 {
-  auto output_count  = std::distance(begin, end);
-  auto strings_count = strings.size();
+  auto const output_count  = std::distance(begin, end);
+  auto const strings_count = strings.size();
   if (output_count == 0) return make_empty_strings_column(stream, mr);
 
-  auto strings_column = column_device_view::create(strings.parent(), stream);
-  auto d_strings      = *strings_column;
-
-  // build offsets column
-  auto offsets_column = make_numeric_column(
+  // allocate offsets column
+  auto out_offsets_column = make_numeric_column(
     data_type{type_id::INT32}, output_count + 1, mask_state::UNALLOCATED, stream, mr);
-  auto d_offsets = offsets_column->mutable_view().template data<int32_t>();
+  auto const d_out_offsets = out_offsets_column->mutable_view().template data<int32_t>();
+
+  if (strings_count == 0) {
```

Comment:
I'm not seeing why this is needed.
The `transform` should put all zeros into `d_out_offsets`
The `reduce` should result in 0 bytes.
The `exclusive_scan` should be a no-op to `d_out_offsets`

Reply:
The problem arises when we try to access `strings.offsets()` as one of the inputs to the transform.  `strings.offsets()` and `strings.chars()` will throw when the strings column is empty.

Reply:
Ok. Could this be avoided by just changing this line?
```
auto const d_in_offsets = strings_count > 0 ? (strings.offsets().data<int32_t>() + strings.offset()) : nullptr;
```

Reply:
Thanks, @davidwendt!  Using `nullptr` for the offsets and chars made it a lot cleaner.